### ISSUE TITLE
Improved `alignParameters` handling should be opt-in

### ISSUE THREAD

**Issue by jkinkead** [scala-ide/scalariform#206]
`alignParameters` handling was expanded in the `daniel-trinh` fork to include aligning equals signs and default values.

This should possibly be hidden behind an option per #142 .

**Comment by jkinkead** [comment]
Should this be milestone 1.0, @sschaef / @fommil ? It's an improvement to an existing opt-in feature, but I notice that @jrudolph disabled `alignParameters` in https://github.com/akka/akka/pull/19936 , probably for this reason.

**Comment by jkinkead** [comment]
To visualize what I'm talking about, here's an example from akka:
```diff
diff --git a/akka-actor/src/main/scala/akka/routing/ConsistentHashing.scala b/akka-actor/src/main/scala/akka/routing/ConsistentHashing.scala
index 1139fc1..e0ba2b1 100644
--- a/akka-actor/src/main/scala/akka/routing/ConsistentHashing.scala
+++ b/akka-actor/src/main/scala/akka/routing/ConsistentHashing.scala
@@ -135,9 +135,9 @@ object ConsistentHashingRoutingLogic {
  */
 @SerialVersionUID(1L)
 final case class ConsistentHashingRoutingLogic(
-  system: ActorSystem,
-  virtualNodesFactor: Int = 0,
-  hashMapping: ConsistentHashingRouter.ConsistentHashMapping = ConsistentHashingRouter.emptyConsistentHashMapping)
+  system:             ActorSystem,
+  virtualNodesFactor: Int                                           = 0,
+  hashMapping:        ConsistentHashingRouter.ConsistentHashMapping = ConsistentHashingRouter.emptyConsistentHashMapping)
   extends RoutingLogic {

   import ConsistentHashingRouter._
```
Note that the types are aligned as are the default values. This is a new addition to the `alignParameters` setting - previously, it left-justified all arguments, but left the types & defaults alone.

**Comment by fommil** [comment]
I really like alignment but haven't used it yet.

**Comment by jkinkead** [comment]
It's more that you can no longer opt-in to formatting like:
```scala
// Int is not aligned with String.
def foo(arg1: Int,
        arg2HasALongName: String): Unit
```
This is the kind of alignment that v0.1.4 does. Akka disabled `alignParameters` in the PR I referenced, likely because of this issue.

**Comment by jkinkead** [comment]
Attached is the full Akka diff with all of the new settings from PR #205 . Note that virtually all of these are either arrow renaming changes or the new align parameters setting. FYI @jrudolph if you're interested in comparing with your PR.

[akkadiff.txt](https://github.com/scala-ide/scalariform/files/154048/akkadiff.txt)

**Comment by fommil** [comment]
:smile: wow, we're actually getting near 1.0

**Comment by jrudolph** [comment]
I tried your branch with `.setPreference(FirstParameterOnNewline, Preserve)` but it helps only a little because of these kind of changes:

```diff
   def validate[T1, T2](value1: RequestVal[T1],
-                       value2: RequestVal[T2],
-                       check: Function2[T1, T2, JBoolean],
-                       errorMsg: String,
-                       innerRoute: Route, moreInnerRoutes: Route*): Route =
+    value2: RequestVal[T2],
+    check: Function2[T1, T2, JBoolean],
+    errorMsg: String,
+    innerRoute: Route, moreInnerRoutes: Route*): Route =
```

So, the first line is now preserved but the indentation of the others still changes.

**Comment by jkinkead** [comment]
@jrudolph - It looks like you left the `.setPreference(AlignParameters, false)` change in. Try setting that back to `true`. :)

**Comment by jrudolph** [comment]
Setting it to `true` does more than we need:

```diff
-  def validate[T1, T2](value1: RequestVal[T1],
-                       value2: RequestVal[T2],
-                       check: Function2[T1, T2, JBoolean],
-                       errorMsg: String,
+  def validate[T1, T2](value1:     RequestVal[T1],
+                       value2:     RequestVal[T2],
+                       check:      Function2[T1, T2, JBoolean],
+                       errorMsg:   String,
```

**Comment by jkinkead** [comment]
Precisely. That's what this bug intends to address: The additional alignment of the types & default values in method signatures should be behind a preference, instead of the default behavior of `alignParameters`.

So, it sounds like this should be milestone 1.0 - @fommil / @sschaef ? I'd tag it myself, but I don't have permissions. :)

**Comment by fommil** [comment]
we should probably get you added

**Comment by jkinkead** [comment]
FYI https://github.com/jkinkead/scalariform/tree/align_parameter_types_tmp has a fix. It's very simple, but it's dependent on #205 .

**Comment by jrudolph** [comment]
@jkinkead thanks, makes sense. If I get to it I'll try your branch.

**Comment by jrudolph** [comment]
Cool, your new switch works nicely. What's now left over for Akka is
 * RARROW
 * implicit parameter list formatting (you seem to have a suggestion in #180)
 * it seems to delete lines from block comments, e.g. here https://github.com/akka/akka/pull/19936/files#diff-a09ffa60a39c454ac5f877fda4826a03L21

**Comment by jkinkead** [comment]
Awesome, that list aligns with my observations as well. :)

* RARROW is tracked in #209 . I'd like some confirmation that people are OK just removing the formatting, instead of hiding it behind a flag (it's dangerous).
* Implicit auto-formatting I can move to a flag.
* Third issue is a regression, fixed in #208 .
